src/transactions/tippedVote.js

```diff
@@ -0,0 +1,116 @@
+module.exports = {
+    fields: ['link', 'author', 'vt', 'tag', 'tip'],
+    validate: (tx, ts, legitUser, cb) => {
+        // Check if author vote exists and whether author has claimed the reward
+        if (!validate.string(tx.data.author, config.accountMaxLength, config.accountMinLength, config.allowedUsernameChars, config.allowedUsernameCharsOnlyMiddle))
+            return cb(false, 'invalid tx data.author')
+        
+        if (!validate.string(tx.data.link, config.accountMaxLength, config.accountMinLength))
+            return cb(false, 'invalid tx data.link')
+        
+        if (!validate.integer(tx.data.vt, false, true))
+            return cb(false, 'invalid tx data.vt')
+        
+        if (!validate.string(tx.data.tag, config.tagMaxLength))
+            return cb(false, 'invalid tx data.tag')
+        
+        if (!transaction.hasEnoughVT(tx.data.vt, ts, legitUser))
+            return cb(false, 'invalid tx not enough vt')
+
+        // tip should be between 1 and 10^config.tippedVotePrecision
+        if (!validate.integer(tx.data.tip,false,false,Math.pow(10,config.tippedVotePrecision),1))
+            return cb(false, 'invalid author tip value')
+        
+        cache.findOne('contents', {_id: tx.data.author+'/'+tx.data.link}, (err, content) => {
+            if (err) throw err
+            if (!content) return cb(false, 'cannot vote and tip non-existent content')
+            if (content.votes.length === 0) return cb(false, 'no votes in this content to tip author with')
+
+            // author should be voted and not claimed reward
+            let authorVote = false
+            let authorVoteClaimed = false
+            for (let v = 0; v < content.votes.length; v++) {
+                if (content.votes[v].u === tx.data.author) {
+                    authorVote = true
+                    if (content.votes[v].claimed)
+                        authorVoteClaimed = true
```

Comment:
Couldn't it be replaced by:
authorVoteClaimed = content.votes[v].claimed
So we don't check a thing to make another thing have the same value.
By the way, I don't know the default value of content.votes[v].claimed and what else it could be.

Reply:
The `claimed` value in the votes array stores the timestamp of the reward claim time. Setting it to that timestamp instead of a boolean does not make any difference imo

Reply:
So, it's my fault, I think that is better as is now, I had supposed that claimed had only true on a claimed value, but a timestamp is different, because we don't need unnecessary complexity that could depend on something that could change from a nodejs version to another.

I'm referring also on problems like this one: https://stackoverflow.com/questions/12236271/is-true-1-and-false-0-in-javascript/12236341#12236341